tests/test_atproto_client/models/tests/test_string_formats.py

```diff
@@ -0,0 +1,221 @@
+
+# Test data sourced directly from bluesky-social/atproto repo:
+# https://github.com/bluesky-social/atproto/tree/main/interop-test-files/syntax
+INTEROP_TEST_FILES_DIR: Path = Path('tests/test_atproto_client/interop-test-files/syntax')
+
+
+# TODO: 230 passed, 11 xfailed
+# These cases appear in both _valid.txt and _invalid.txt files.
+# Need investigation to determine if our validation is incorrect or if test data needs updating:
+SKIP_THESE_VALUES = [
+    (
+        string_formats.AtUri,
+        'at://did:plc:asdf123',
+    ),  # Listed as both valid and invalid in AT-URI files under "enforces spec basics"
+    (
+        string_formats.AtUri,
+        'at://did:plc:asdf123/com.atproto.feed.post',
+    ),  # Same AT-URI pattern - appears in both valid/invalid files
+    (
+        string_formats.DateTime,
+        '1985-04-12T23:20:50.123Z',
+    ),  # Listed as "preferred" in valid but also appears in invalid under RFC-3339 section
+    (
+        string_formats.DateTime,
+        '1985-04-12T23:20:50.123-00:00',
+    ),  # Listed as "supported" in valid but marked invalid under timezone formats
+    (string_formats.DateTime, '1985-04-12T23:20Z'),  # Similar timezone format discrepancy between valid/invalid files
+    (string_formats.Handle, 'john.test'),  # Base pattern appears valid but numeric suffix versions are marked invalid
+    (string_formats.Nsid, 'one.two.three'),  # Same pattern - base form valid but numeric suffixes marked invalid
+]
+
+
+def get_test_cases(filename: str) -> List[str]:
+    """Get non-comment, non-empty lines from an interop test file.
+
+    Important: Preserves whitespace in test cases. This is critical for
+    format validators where leading/trailing/internal whitespace makes a
+    value invalid. For example, ' 1985-04-12T23:20:50.123Z' (with leading space)
+    should be invalid for datetime validation.
+
+    Args:
+        filename: Name of the test file to read from interop test files directory
+
+    Returns:
+        List of test cases with original whitespace preserved
+    """
+    return [
+        line
+        for line in INTEROP_TEST_FILES_DIR.joinpath(filename).read_text().splitlines()
+        if line and not line.startswith('#')
+    ]
+
+
+@pytest.fixture
+def valid_handles() -> List[str]:
+    return get_test_cases('handle_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_dids() -> List[str]:
+    return get_test_cases('did_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_nsids() -> List[str]:
+    return get_test_cases('nsid_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_aturis() -> List[str]:
+    return get_test_cases('aturi_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_datetimes() -> List[str]:
+    return get_test_cases('datetime_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_tids() -> List[str]:
+    return get_test_cases('tid_syntax_valid.txt')
+
+
+@pytest.fixture
+def valid_record_keys() -> List[str]:
+    return get_test_cases('recordkey_syntax_valid.txt')
```

Comment:
should we delete these fixtures? as I can see we are using it only once in the `valid_data` fixture. and for `invalid_data` we have different approach. we call directly `get_test_cases` without fixture per each. so i want to pick 1 way to do the same (i prefer not to create a lot of fixtures per each file)

Reply:
ahh my fault for being sloppy there - updated to isolate all the io and removed all unused fixtures